nodes.py

```diff
@@ -1421,38 +1437,91 @@ def register():
         generate_node_type(prefs, name, ET.parse(arg_file).getroot())
 
     pattern_nodeitems = []
+    pattern_texture_nodeitems = []
+    pattern_bump_nodeitems = []
+    pattern_color_nodeitems = []
+    pattern_manifold_nodeitems = []
+    pattern_geometry_nodeitems = []
+    pattern_script_nodeitems = []
     bxdf_nodeitems = []
+    lmbxdf_nodeitems = []
     light_nodeitems = []
     displacement_nodeitems = []
+    depreciated_nodeitems = []
+    
     for name, node_type in RendermanPatternGraph.nodetypes.items():
         node_item = NodeItem(name, label=node_type.bl_label)
-        if node_type.renderman_node_type == 'pattern':
-            pattern_nodeitems.append(node_item)
-        elif node_type.renderman_node_type == 'bxdf':
-            bxdf_nodeitems.append(node_item)
-        elif node_type.renderman_node_type == 'light':
-            light_nodeitems.append(node_item)
-        elif node_type.renderman_node_type == 'displacement':
-            displacement_nodeitems.append(node_item)
+        if CheckPattern(name, "depreciated") == 1:
+            depreciated_nodeitems.append(node_item)
+        else:
+            if node_type.renderman_node_type == 'pattern':
+                if CheckPattern(name, "texture") == 1:
```

Comment:
Damn code comments finally show up after you merge it... returning the type would be useful if we weren't already using the semi-autonomous if statement block, as it is now all that buys us is slightly more memory usage since we are using a string as opposed to an int (if python has similar memory conservation rules as Java or C)